supervision/detection/utils.py

```diff
@@ -1039,3 +1039,60 @@ def cross_product(anchors: np.ndarray, vector: Vector) -> np.ndarray:
     )
     vector_start = np.array([vector.start.x, vector.start.y])
     return np.cross(vector_at_zero, anchors - vector_start)
+
+
+def get_box_intersection(
+    xyxy_1: np.ndarray, xyxy_2: np.ndarray
+) -> Optional[np.ndarray]:
+    overlap_xmin = max(xyxy_1[0], xyxy_2[0])
+    overlap_ymin = max(xyxy_1[1], xyxy_2[1])
+    overlap_xmax = min(xyxy_1[2], xyxy_2[2])
+    overlap_ymax = min(xyxy_1[3], xyxy_2[3])
+
+    if overlap_xmin < overlap_xmax and overlap_ymin < overlap_ymax:
+        return np.array([overlap_xmin, overlap_ymin, overlap_xmax, overlap_ymax])
+    else:
+        return None
+
+
+def get_unit_vector(xy_1: np.ndarray, xy_2: np.ndarray) -> np.ndarray:
+    direction = xy_2 - xy_1
+    magnitude = np.linalg.norm(direction)
+    return direction / magnitude if magnitude > 0 else np.zeros(2)
+
+
+def spread_out_boxes(
+    xyxy: np.ndarray, step: int, max_iterations: int = 100
+) -> np.ndarray:
+    if len(xyxy) == 0:
+        return xyxy
+
+    xyxy_padded = pad_boxes(xyxy, px=step)
+    for _ in range(max_iterations):
+        iou = box_iou_batch(xyxy_padded, xyxy_padded)
+        np.fill_diagonal(iou, 0)
+
+        if np.all(iou == 0):
+            break
+
+        i, j = np.unravel_index(np.argmax(iou), iou.shape)
+
+        xyxy_i, xyxy_j = xyxy_padded[i], xyxy_padded[j]
+        box_intersection = get_box_intersection(xyxy_i, xyxy_j)
+        assert (
+            box_intersection is not None
+        ), "Since we checked IoU already, boxes should always intersect"
+
+        intersection_center = (box_intersection[:2] + box_intersection[2:]) / 2
+        xyxy_i_center = (xyxy_i[:2] + xyxy_i[2:]) / 2
+        xyxy_j_center = (xyxy_j[:2] + xyxy_j[2:]) / 2
+
+        unit_vector_i = get_unit_vector(intersection_center, xyxy_i_center)
+        unit_vector_j = get_unit_vector(intersection_center, xyxy_j_center)
+
+        xyxy_padded[i, [0, 2]] += int(unit_vector_i[0] * step)
```

Comment:
We have `move_boxes` util. Looks like we could replace:

```python
xyxy_padded[i, [0, 2]] += int(unit_vector_i[0] * step)
xyxy_padded[i, [1, 3]] += int(unit_vector_i[1] * step)
```

With:

```python
offset=(unit_vector_i * step).astype(int)
xyxy_padded = move_boxes(xyxy=xyxy_padded, offset=offset)
```

I find it a lot more readable.

Reply:
I didn't think about it - good point.

However, the current algo moves one box at a time, and your suggestion would be part of the vectorized approach.

My plan for this was to not entirely rewrite the contributor's code, but to open a second issue for the community, adding vectorization as a second step.